packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,237 @@
+     * @param newL1WithdrawContract address of the new L1 withdrawContract.
+     */
+    function changeL1WithdrawContract(address newL1WithdrawContract)
+        public
+        onlyFromCrossDomainAccount(l1WithdrawContract)
+    {
+        emit L1WithdrawContractChanged(l1WithdrawContract, newL1WithdrawContract);
+        l1WithdrawContract = newL1WithdrawContract;
+    }
 
-    function disableDepositContract() public onlyFromCrossDomainAccount(l1Owner) {}
+    /**
+     * @notice Enables L1 owner to whitelist a L1 Token <-> L2 Token pair for bridging.
+     * @dev Only callable by the existing l1WithdrawContract via the optimism cross domain messenger.
+     * @param l1Token Address of the canonical L1 token. This is the token users will receive on Ethereum.
+     * @param l2Token Address of the L2 token representation. This is the token users would deposit on optimism.
+     */
+    function whitelistToken(address l1Token, address l2Token) public onlyFromCrossDomainAccount(l1WithdrawContract) {
+        whitelistedTokens[l2Token] = WhitelistedToken({ l1Token: l1Token, lastBridgeTime: uint64(getCurrentTime()) });
 
-    function transferL1Owner(address _l1Owner) public onlyFromCrossDomainAccount(l1Owner) {
-        l1Owner = _l1Owner;
+        emit TokenWhitelisted(l1Token, l2Token);
     }
 
-    // Depositor functions
+    /**
+     * @notice L1 owner can enable/disable deposits over all whitelisted tokens.
+     * @dev Only callable by the existing l1WithdrawContract via the optimism cross domain messenger.
+     * @param _depositsEnabled bool to set if the deposit box should accept/reject deposits.
+     */
+    function setEnableDeposits(bool _depositsEnabled) public onlyFromCrossDomainAccount(l1WithdrawContract) {
+        depositsEnabled = _depositsEnabled;
+        emit DepositsEnabled(_depositsEnabled);
+    }
 
+    /**************************************
+     *         DEPOSITOR FUNCTIONS        *
+     **************************************/
+
+    /**
+     * @notice Called by L2 user to bridge funds between L2 and L1.
+     * @dev Emits the `L2FundsDeposited` event which relayers listen for as part of the bridging action.
+     * @dev The caller must first approve this contract to spend `amount` of `l2Token`.
+     * @param recipient L1 address that should receive the tokens.
+     * @param l2Token L2 token to deposit.
+     * @param amount How many L2 tokens should be deposited.
+     * @param maxFee Max fraction of the total `amount` that the depositor is willing to pay as a fee. scaled by 1e18.
+     */
     function deposit(
         address recipient,
-        address l1Token,
+        address l2Token,
         uint256 amount,
         uint256 maxFee
-    ) public {}
+    ) public onlyIfDepositsEnabled() {
+        require(isTokenWhitelisted(l2Token), "deposit token not whitelisted");
+        require(maxFee <= 1e18, "max fee can not be over 1e18");
```

Comment:
why this setting? what if precision is 6 decimals?

Reply:
this represents a fraction of `amount` of the token that will be sent as a fee. i.e If I'm willing to take a 5% fee this will be 0.05 or 0.05e18. by definition, I cant take more than 100% as a fee (or 1e18) in the context of this scalling.

Reply:
Ah ok this was definitely not clear. Would recommend you declare a constant ONE_HUNDRED_PCT = 1e18 at top of file and then rename `maxFee` to `maxFeePct`

Reply:
ok I'll change it to `maxFeePct`. would really rather not define the `ONE_HUNDRED_PCT` const though; feels like that is excessive. I'll change the require message to make it more clear.

Reply:
SGTM